src/utils/queryBuilder.ts

```diff
@@ -84,8 +85,70 @@ const buildBestScoreSort = (): ISortQuery => ({
   },
 });
 
-const buildSearchQuery = (searchFieldsObject: ISearchPayload, fieldsToSearch: string[] = []): IQuery => {
-  const { fields, sort, rowsPerPage } = searchFieldsObject;
+const buildGeoShapeQuery = (north, south, east, west, depth) => {
+  const geoShape: IShapeCoordinates = {
+    type: 'envelope',
+    coordinates: [
+      [parseFloat(west), parseFloat(north)],
+      [parseFloat(east), parseFloat(south)],
+    ],
+  };
+
+  const geoShapeQuery: IGeoShapeQuery = {
+    geo_shape: {
+      geom: {
+        shape: geoShape,
+        relation: 'intersects',
+      },
+    },
+  };
+
+  if (depth && geoShapeQuery.geo_shape.geom) {
+    geoShapeQuery.geo_shape.geom.depth = {
+      from: 0,
+      to: parseInt(depth),
+    };
+  }
+  return geoShapeQuery;
+};
+
+const buildSortQuery = (finalQuery, sort, isCount) => {
+  if (sort && !isCount) {
+    const sortQuery: ISortQuery =
+      sort === 'recent_study' ? buildCustomSortScriptForStudyPeriod() : buildBestScoreSort();
+    finalQuery.sort?.push(sortQuery);
+  } else {
+    delete finalQuery.sort;
+  }
+  return finalQuery;
+};
+
+const buildAggregationQuery = (finalQuery, ignoreAggregation, isCount, aggregationField) => {
+  if (!ignoreAggregation && !isCount && aggregationField) {
+    const aggregateQuery: IAggregateQuery = {
+      unique_values: {
+        terms: {
+          field: aggregationField,
+        },
+      },
+    };
+    finalQuery.aggs = aggregateQuery;
+    finalQuery.size = 0;
```

Comment:
While fetching aggregated unique values for filters, we don't need records. Sending size as 0 can prevent us from fetching records in elasticsearch. We require only aggregated unique values, so by sending size as 0, we can reduce the response size. This will only occur when fetching values from the filter dropdown, not in other cases.